Tests/OktaOAuth2Tests/JWTAuthorizationFlowTests.swift

```diff
@@ -0,0 +1,165 @@
+//
+// Copyright (c) 2024-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import XCTest
+@testable import TestCommon
+@testable import AuthFoundation
+@testable import OktaOAuth2
+
+final class JWTAuthorizationFlowDelegateRecorder: AuthenticationDelegate {
+    typealias Flow = JWTAuthorizationFlow
+    
+    var token: Token?
+    var error: OAuth2Error?
+    var url: URL?
+    var started = false
+    var finished = false
+    
+    func authenticationStarted<Flow>(flow: Flow) {
+        started = true
```

Comment:
should we record the flow?

Reply:
This DelegateRecorder class is used in testing, to observe invocations of the authentication delegate, and to validate that the appropriate calls are made, and in the correct order.

I really need to make this class support generics so it can be reused elsewhere (e.g. there's almost identical copies of this for Resource Owner Flow, Token Exchange Flow, etc), but I didn't feel that was a priority at this time.